ord_schema/message_helpers.py

```diff
@@ -253,36 +253,89 @@ def find_submessages(message, submessage_type):
     return submessages
 
 
-def get_compound_mol(compound):
-    """Retrieves an RDKit molecule object from a Compound.
+def smiles_from_compound(compound):
+    """Fetches or generates a SMILES identifier for a compound.
+
+    If a SMILES identifier already exists, it is simply returned.
 
     Args:
         compound: reaction_pb2.Compound message.
 
     Returns:
-        RDKit molecule.
+        smiles: SMILES identifier.
     """
-    if not Chem:
-        raise ImportError('Missing RDKit dependency')
     for identifier in compound.identifiers:
-        if identifier.type == reaction_pb2.CompoundIdentifier.RDKIT_BINARY:
-            return Chem.Mol(identifier.bytes_value)
-    return None
+        if identifier.type == reaction_pb2.CompoundIdentifier.SMILES:
+            return identifier.value
+    return Chem.MolToSmiles(mol_from_compound(compound))
 
 
-def get_compound_smiles(compound):
-    """Retrieves a SMILES string identifier from a Compound.
+# pylint: disable=inconsistent-return-statements
+def mol_from_compound(compound, return_identifier=False):
+    """Creates an RDKit Mol from a Compound message.
+
+    Note that this function prefers RDKIT_BINARY identifiers over any others.
 
     Args:
         compound: reaction_pb2.Compound message.
+        return_identifier: If True, return the CompoundIdentifier used to
+            create the Mol.
 
     Returns:
-        The SMILES string identifier for the compound or None.
+        mol: RDKit Mol.
+        identifier: The identifier that was used to create `mol`. Only returned
+            if `return_identifier` is True.
+
+    Raises:
+        ValueError: If no structural identifier is available, or if the
+            resulting Mol object is invalid.
     """
+    # Prefer RDKIT_BINARY identifiers when available.
     for identifier in compound.identifiers:
-        if identifier.type == reaction_pb2.CompoundIdentifier.SMILES:
-            return identifier.value
-    return None
+        if identifier.type == reaction_pb2.CompoundIdentifier.RDKIT_BINARY:
+            mol = Chem.Mol(identifier.bytes_value)
+            if return_identifier:
+                return mol, identifier
+            return mol
+    for identifier in compound.identifiers:
+        if identifier.type in _COMPOUND_IDENTIFIER_LOADERS:
+            mol = _COMPOUND_IDENTIFIER_LOADERS[identifier.type](
+                identifier.value)
+            if not mol:
+                raise ValueError(
+                    f'no valid structural identifier for Compound: {compound}')
+            if return_identifier:
+                return mol, identifier
+            return mol
+
+
+# pylint: enable=inconsistent-return-statements
+
+
+def check_compound_identifiers(compound):
+    """Verifies that structural compound identifiers are consistent.
```

Comment:
We might have issues with tautomers here. Roundtripping SMILES -> InChI -> SMILES can change the string. So if someone is programmatically defining a compound and uses both a `MolToSmiles` and `MolToInchi`, this check might fail. This might be worth a quick email to Greg to ask for advice on checking consistency (or not)

Reply:
Agreed; we don't necessarily have to use this anywhere until it's doing what we want---would you prefer that I remove it for now?

Reply:
Where were you planning to use it? I think it would be appropriate to use it when validating a compound to issue a warning, but not an error.

Reply:
I was thinking of putting it into the validation for Compound; we could issue a warning in that case.

Reply:
Added a commit to do this.